scm/github/repo.go

```diff
@@ -662,3 +674,132 @@ func (c *client) GetBranch(ctx context.Context, r *api.Repo, branch string) (str
 
 	return data.GetName(), data.GetCommit().GetSHA(), nil
 }
+
+// GetNetrcPassword returns a clone token using the repo's github app installation if it exists.
+// If not, it defaults to the user OAuth token.
+func (c *client) GetNetrcPassword(ctx context.Context, db database.Interface, r *api.Repo, u *api.User, g yaml.Git) (string, error) {
+	l := c.Logger.WithFields(logrus.Fields{
+		"org":  r.GetOrg(),
+		"repo": r.GetName(),
+	})
+
+	l.Tracef("getting netrc password for %s/%s", r.GetOrg(), r.GetName())
+
+	var err error
+
+	// repos that the token has access to
+	// providing no repos, nil, or empty slice will default the token permissions to the list
+	// of repos added to the installation
+	repos := g.Repositories
+
+	// use triggering repo as a restrictive default
+	if repos == nil {
+		repos = []string{r.GetName()}
+	}
+
+	// convert repo fullname org/name to just name for usability
+	for i, repo := range repos {
+		split := strings.Split(repo, "/")
+		if len(split) == 2 {
+			repos[i] = split[1]
+		}
+	}
+
+	// permissions that are applied to the token for every repo provided
+	// providing no permissions, nil, or empty map will default to the permissions
+	// of the GitHub App installation
+	//
+	// the Vela compiler follows a least-privileged-defaults model where
+	// the list contains only the triggering repo, unless provided in the git yaml block
```

Comment:
I'm a little confused at this comment because the comment earlier in this function says
```
	// providing no repos, nil, or empty slice will default the token permissions to the list
	// of repos added to the installation
```

But then this one says that it will only contain the triggering repo. Am I getting confused?